gallery/src/DarkModeModal/DarkModeModal.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+
+import {
+  NxButton,
+  NxModal,
+  NxForm,
+  NxFormGroup,
+  NxToggle,
+  useToggle,
+  NxFieldset,
+  NxRadio,
+  NxTooltip,
+  NxFontAwesomeIcon
+} from '@sonatype/react-shared-components';
+
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+
+const DarkModeModal = () => {
+  const [showModal, setShowModal] = useState(false);
+  const modalCloseHandler = () => setShowModal(false);
+
+  const [enableModeChange, toggleEnableModeChange] = useToggle(false);
+  const [mode, setMode] = useState<string | null>(null);
+
+  const { classList } = document.documentElement;
+
+  useEffect(() => {
+    classList.toggle('nx-html--enable-color-schemes', enableModeChange);
+  }, [enableModeChange]);
+
+  useEffect(() => {
+    classList.toggle('nx-html--dark-mode', mode === 'dark' && enableModeChange);
+    classList.toggle('nx-html--light-mode', mode === 'light' && enableModeChange);
+  }, [enableModeChange, mode]);
+
+  return (
+    <>
+      <NxButton variant="tertiary" onClick={() => setShowModal(true)}>Theme Settings</NxButton>
```

Comment:
It's a bit atypical to have the button that opens the modal be in the same component as the modal.  Let's move the button itself and the state it controls to the `PageHeader` component

Reply:
> It's a bit atypical to have the button that opens the modal be in the same component as the modal. Let's move the button itself and the state it controls to the `PageHeader` component

Sorry, this one came with a lot of changes. With the button moving to `PageHeader`, the settings inside the modal reset each time it was opened. So there is now an additional useEffect to check if various classnames are present, and set the mode accordingly. I removed state that controlled the toggling of `enableModeChange`, in favour of one `mode` state that has a `disabled` option as well.  [c5cda2c](https://github.com/sonatype/sonatype-react-shared-components/pull/782/commits/c5cda2ca1dc81c9a3682c94dc0073bf3c1f749e0)